### PR TITLE
Some Graph Problems Added

### PR DESCRIPTION
**Pull request by DBasu2610**
I added BFS, DFS, Dijkstra and Bellman-Ford in java

### CODE REVIEW

**Comment by iamwatchdogs** [comment]
Hi @DBasu2610, Please make sure you pass the linting check so that I can review it and merge the PR. You can find some help from review dog on [file changes](https://github.com/Grow-with-Open-Source/DSA/pull/6/files) section and by reviewing the [Displaying Diffs](https://github.com/Grow-with-Open-Source/DSA/actions/runs/11081741771/job/30819990272?pr=6) section in the output workflow.

> [!NOTE]
> If you don't see any remarks on your code in the file changes section, make sure to turn on the annotations.
> image

**Comment by iamwatchdogs** [comment]
@DBasu2610 I went thought you changes and I feel like you have mistaken my previous comment. I really appreciate you putting more effort about learning about review dog and trying to create a workflow from scratch for this repo, but you're fairly mistaken.

The CI workflow _(linting workflows, to be specific)_ are implemented in such a way to help you find the right way to better format your code. And for your situation, the linter used for your Java source code uses review dog internally. I just pointed you that there were few linting checks that needed to be passed so that this PR could get merge. And I directed you to source that can help you format your source code so that your source code can get pass the liniting checks.

The review dog was supposed to show the part where the formatting when wrong and the linting workflow failed. All you had to do was goto the `Files Changes` section within this PR. Just as show below,

| image |
| --- |
| image |

And I have also pointed you to the direct solution from the output log from the linting workflow by saying, **"by reviewing the diffs"**,

| image |
| --- |
| image |
| image |

I really hope this clears the miscommunications we just had and helps revise your actual PR submission.

**Comment by iamwatchdogs** [comment]
Now that we're clear with our miscommunications, let's take a bit about the fixing things...

To start off, I suggest you to undo you recent changes i.e., building a npm env for review dog and creating a workflow for the review dog,

```bash
# reset your last two commits
git reset --hard HEAD~2

# foce push to your remote repo for the changes to reflect here
git push origin main -f
```

Now, with the help of the above mentioned source, format your source code, then commit & push them to here. So that we get pass linting checks and get your PR merged.

```bash
# after making your required changes
git add .

git commit -m "<proper-message>"

git push origin main
```

> [!IMPORTANT]
> Please follow the proper file structure provided to you from the beginning. In other words, maintain consistency of the file structure by placing in specific hierarchy i.e., not in `SolvedProblems` but in `Solved-Problems`.

> [!TIP]
> Please do ping me up for any clarifications and please don't act upon assumptions. I'm always happy to help you out in clarifying things and discussing for better solutions. Thank you for your interest and patience...

**Comment by iamwatchdogs** [comment]
Hi @DBasu2610, You're making great progress and did follow all of my instruction. But there are still some liniting checks needed to be passed. Please try to format your source code so that I could merge this PR.

If you find the review dogs's annotation on `File Changes` confusing, you can checkout the formatted version of the code on  second source i.e., [reviewing the diffs](https://github.com/Grow-with-Open-Source/DSA/actions/runs/11100529665/job/30836591974?pr=6).

image

Please do ping if you're facing issue, even after that...

File: Solved-Problems/GraphTheory/DFS.java
Comment (iamwatchdogs) [inline Solved-Problems/GraphTheory/DFS.java:1]: This is the file that's causing the linting to fail, please format the code with provided sources.

Comment (iamwatchdogs) [reply Solved-Problems/GraphTheory/DFS.java:1]: There was no need to remove you source code @DBasu2610, you could have formatted it. The linting checks failed due to indentation and an missing curly brace. And I directed right towards the answer i.e., [reviewing diffs](https://github.com/Grow-with-Open-Source/DSA/actions/runs/11100981942/job/30840392797?pr=6). It has the formatted code.

I know the linting workflow might annoy you but they're the ones that help you write clean code following standards. Even after hint the direct solution, you proceed to remove the file and I'm saying it was unnecessary. You could have copy-pasted the solution, removing things won't make it any better.

The solution was here along... You could have asked me even for clarifications, but yet you proceeded on your own...
image

I gave you the solution and it's up to you whether to use it or not...

File: Solved-Problems/GraphTheory/BellmanFord.java
Comment (iamwatchdogs) [inline Solved-Problems/GraphTheory/BellmanFord.java:1]: Now this file is failing the linting checks. Please use to review dog's annotations or diffs from the workflow to resolve the linting checks...

**Comment by iamwatchdogs** [comment]
Please refer to on of the outdate comment on file `DFS.java`.

<https://github.com/Grow-with-Open-Source/DSA/pull/6#discussion_r1780645689>

**Comment by iamwatchdogs** [comment]
@DBasu2610 You are doing great by learning how to format your code, but as I said [earlier](https://github.com/Grow-with-Open-Source/DSA/pull/6#discussion_r1780645689) all the solutions are given to your... All the formatted code is present within *"Showing Diffs & exit"* section on the [linting workflow](https://github.com/Grow-with-Open-Source/DSA/actions/runs/11102440133/job/30842274205), so please check out for file too...

File: Solved-Problems/GraphTheory/DFS.java
Comment (DBasu2610) [reply Solved-Problems/GraphTheory/DFS.java:1]: Yes I added it back and also formatted it, but it still is having linting issues

File: Solved-Problems/GraphTheory/README.md
Comment (iamwatchdogs) [inline Solved-Problems/GraphTheory/README.md:1]: @DBasu2610 This README.md file contains the same content from the README.md from `Solved-Problems` directory. And I want to remind you that, this file is the file that will be rendered on the web page for this repo. So, make sure to describe the question for your solution or remove this file.

Here's an example, 
[Backspace_String_Compare](https://github.com/Grow-with-Open-Source/DSA/tree/main/Solved-Problems/Backspace_String_Compare) was a solution that got merged into this repo and here the [web page](https://grow-with-open-source.github.io/DSA/Solved-Problems/Backspace_String_Compare/) for this solution. I hope this give you a clear idea of how this repo is designed.

File: Solved-Problems/GraphTheory/index.md
Comment (iamwatchdogs) [inline Solved-Problems/GraphTheory/index.md:1]: Please remove this file completely.

Comment (DBasu2610) [reply Solved-Problems/GraphTheory/index.md:1]: removed

File: Solved-Problems/GraphTheory/DFS.java
Comment (iamwatchdogs) [reply Solved-Problems/GraphTheory/DFS.java:1]: Look here, there are the things that need to changes...

| image |
| --- |
| image |
| image |

Comment (iamwatchdogs) [reply Solved-Problems/GraphTheory/DFS.java:1]: Try directly copying from the diffs and pasting them right in the GitHub. That should work...

**Comment by DBasu2610** [comment]
I redid the format of DFS.java.

**Comment by iamwatchdogs** [comment]
Please refer back to this [conversation](https://github.com/Grow-with-Open-Source/DSA/pull/6#discussion_r1780690345).

**Comment by iamwatchdogs** [comment]
@DBasu2610, Try copy-pasting the formatted code from [**"Show Diffs and exit"**](https://github.com/Grow-with-Open-Source/DSA/actions/runs/11106553911/job/30858169977?pr=6) as mentioned in the previous [conversation](https://github.com/Grow-with-Open-Source/DSA/pull/6#discussion_r1780694660).

**Comment by DBasu2610** [comment]
I did copy the formatted code from "show diffs and exit" and pasted it directly

File: Solved-Problems/GraphTheory/DFS.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+public class DFS {
+  // 1- --3
+  // / | \
+  // 0 | 5--6
+  // \ | /
+  // 2-- - 4
+
+  public static void main(String[] args) {
+    int vertex = 7;
+    @SuppressWarnings("unchecked") ArrayList<Edge> graph[] = new ArrayList[vertex];
```
Comment (iamwatchdogs) [inline Solved-Problems/GraphTheory/DFS.java:12]: Please fix the usage of brackets

```suggestion
    @SuppressWarnings("unchecked") ArrayList<Edge> [] graph = new ArrayList[vertex];
```

File: Solved-Problems/GraphTheory/DFS.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    boolean vis[] = new boolean[vertex];
```
Comment (iamwatchdogs) [inline Solved-Problems/GraphTheory/DFS.java:13]: Please fix the usage of brackets

```suggestion
    boolean [] vis = new boolean[vertex];
```

File: Solved-Problems/GraphTheory/DFS.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    int src;
+    int dest;
+
+    public Edge(int s, int d) {
+      this.src = s;
+      this.dest = d;
+    }
+  }
+
+  public static void createGraph(ArrayList<Edge> graph[]) {
```
Comment (iamwatchdogs) [inline Solved-Problems/GraphTheory/DFS.java:31]: Please fix the usage of brackets

```suggestion
  public static void createGraph(ArrayList<Edge> [] graph) {
```

File: Solved-Problems/GraphTheory/DFS.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    graph[4].add(new Edge(4, 5));
+
+    graph[5].add(new Edge(5, 3));
+    graph[5].add(new Edge(5, 4));
+    graph[5].add(new Edge(5, 6));
+
+    graph[6].add(new Edge(6, 5));
+  }
+
+  public static void depthFirst(ArrayList<Edge> graph[], int curr, boolean vis[]) {
```
Comment (iamwatchdogs) [inline Solved-Problems/GraphTheory/DFS.java:59]: Please fix the usage of brackets

```suggestion
  public static void depthFirst(ArrayList<Edge> [] graph, int curr, boolean [] vis) {
```

**Comment by DBasu2610** [comment]
Check if it works now, and thank you for being so patient everytime.

**Comment by iamwatchdogs** [comment]
@DBasu2610 You are the more patient one... Many people who have closed it or leave it away... But you're still willing to work on it despite facing this much issues...

**Comment by DBasu2610** [comment]
See, it still didn't work :)

**Comment by iamwatchdogs** [comment]
@DBasu2610 Have you formatted all of the java source code files ?

> Because currently, this is happening...
> ```py
> sum( no_of_linting_checks( [ list of your java files ] ) ) >= error_limit 
> ```

> And formatting all of the java source code file, can do this,
> ```py
> sum( no_of_linting_checks( [ list of your java files ] ) ) < error_limit
> ```

What I'm suggest you is to format all of them, so that cumulatively they don't hit the limit and pass the checks... Please when I mention all of the java source code file it mean [`BFS.java`](https://github.com/Grow-with-Open-Source/DSA/pull/6/files#diff-fc264d1b6adb4f9d322992d564c0e466feb99cbcd3dad5994411c261c04d465a), [`BellmanFord.java`](https://github.com/Grow-with-Open-Source/DSA/pull/6/files#diff-eb002b8254205e6ea0bf60dc7ae191a04823b352b9a71433c849c1e31f2d13e8), [`Dijkstra.java`](https://github.com/Grow-with-Open-Source/DSA/pull/6/files#diff-0695a7a122c8c2462b80d944f4e6a429f5ba9f372d5827201e81b373b6910151).

**Comment by iamwatchdogs** [comment]
Hi @DBasu2610, Sorry for late reply.

Yes, you can close it on your own and yoy have already did. But note that, linting checks might still fail if they are not formatted.

**Pull request closed without merging**